src/sage/combinat/permutation.py

```diff
@@ -7092,7 +7092,20 @@ def _element_constructor_(self, x, check=True):
             [1, 4, 5, 2, 3, 6]
             sage: Permutations(6)(x)            # known bug
             [1, 4, 5, 2, 3, 6]
+
+            check if :trac:`37284` is fixed::
+
+            sage: S5 = SymmetricGroup(5)
+            sage: P5 = Permutations(5)
+            sage: p = S5.list()[3]
+            sage: P5(p)
+            [4, 5, 1, 2, 3]
         """
+        try:
+            if not isinstance(x, Permutation):
+                x = x.domain()
+        except AttributeError:
+            pass
```

Comment:
```suggestion
        if not isinstance(x, Permutation) and hasattr(x, "domain"):
            x = x.domain()
```
You can use `hasattr` to check if `domain` exists. However, from what I understand, `try`-`except` is definitely acceptable in Sage.

Reply:
It is not good to check that something is not a `Permutation`. You should instead check that the input is a `PermutationGroupElement`:
```suggestion
        if isinstance(x, PermutationGroupElement):
            x = x.domain()
```
No accidental catching of things that have a `domain()` that is not a permutation (say, a function from $\{1,2,3\} \to \{0\}$), no `try`-`except` blocks, and simple code.